src/electricity/PretrainDataset.py

```diff
@@ -22,10 +22,12 @@ def __getitem__(self, index):
         status = self.padding_seqs(self.status[start_index:end_index]).copy()
 
         for i in range(len(x)):
-            prob = random.random()
-            if prob <= self.mask_prob:
+            if random.random() <= self.mask_prob:
                 prob = random.random()
-                x[i] = -1 if prob < 0.8 else np.random.normal() if prob < 0.9 else x[i]
+                if prob < 0.8:
+                    x[i] = -1
+                elif prob < 0.9:
+                    x[i] = np.random.normal()
```

Comment:
I have no idea what's happening here, but that's an issue with the original code. The refactored code is at least more readable.

Reply:
I have no idea either why there's a double random check 🤷